extension/healthcheckextension/healthcheckextension.go

```diff
@@ -61,6 +63,9 @@ func (hc *healthCheckExtension) Start(host extension.Host) error {
 }
 
 func (hc *healthCheckExtension) Shutdown() error {
+	if !atomic.CompareAndSwapInt32(&instanceState, instanceCreated, instanceNotCreated) {
+		return errors.New("only a single instance can be created per process")
+	}
```

Comment:
Why does this logic live in `Shutdown` vs `Start` ?

Reply:
I'm not a fan of this at all.  I'm debating refactoring it a bit so the only run one of me logic isn't duplicated

Reply:
ok so we want to clear the flag on shutdown though, not start.  It's indicating whether the instance has been shutdown.